### PR TITLE
Update how-to-map-hresults-and-exceptions.md

### LINKED ISSUE #18615: Another vapor-constant and class?
The table on this page maps the HRESULT identified by the `COR_E_WEAKREFERENCE` preprocessor symbol to the .NET `WeakReferenceException` exception class.

I can't find any definition for that preprocessor symbol (either by searching the SDK header files, or by searching online sites and source control repositories.

I can't find any evidence that a `WeakReferenceException` class exists in .NET.

Again, I can find references to both of these online, but only to the extent that they are mentioned as mapping to one another (so probably copied from this page), but never with any value for `COR_E_WEAKREFERENCE` or additional namespace information for the WeakReferenceException type.

It's interesting to note that when the `COR_E_WEAKREFERENCE` symbol appears in source files (among a list of other HRESULT constants), it is _always_ commented out. No one else appears to be able to identify an actual value for this supposed constant.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2763c768-604b-2d58-7d5f-048a95657ebe
* Version Independent ID: c9755c11-4c6a-9ae6-9e74-99f52a177ff9
* Content: [How to: Map HRESULTs and Exceptions](https://docs.microsoft.com/en-us/dotnet/framework/interop/how-to-map-hresults-and-exceptions)
* Content Source: [docs/framework/interop/how-to-map-hresults-and-exceptions.md](https://github.com/dotnet/docs/blob/master/docs/framework/interop/how-to-map-hresults-and-exceptions.md)
* Product: **dotnet-framework**
* GitHub Login: @gewarren
* Microsoft Alias: **dotnetcontent**

### LINKED ISSUE #18616: Another non-existent constant and .NET class?
This time it's the mapping of HRESULT constant (preprocessor symbol) `COR_E_VTABLECALLSNOTSUPPORTED` and .NET exception class `VTableCallsNotSupportedException`.

I can't find any definition for either symbol or class, either by searching the SDK header files, or by searching online sites and source control repositories.

I can't find any evidence that a `VTableCallsNotSupportedException` class exists in .NET.

Once again, I do find a few references to both of these online, but only to the extent that they are mentioned as mapping to one another (so probably copied from this page), but never with any value for `COR_E_VTABLECALLSNOTSUPPORTED` or additional namespace information for the VTableCallsNotSupportedException type.

When `COR_E_VTABLECALLSNOTSUPPORTED` does appear in a source file (among a list of other HRESULT constants), it is _always_ commented out. No one else appears to be able to identify an actual value for this supposed constant either.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2763c768-604b-2d58-7d5f-048a95657ebe
* Version Independent ID: c9755c11-4c6a-9ae6-9e74-99f52a177ff9
* Content: [How to: Map HRESULTs and Exceptions](https://docs.microsoft.com/en-us/dotnet/framework/interop/how-to-map-hresults-and-exceptions)
* Content Source: [docs/framework/interop/how-to-map-hresults-and-exceptions.md](https://github.com/dotnet/docs/blob/master/docs/framework/interop/how-to-map-hresults-and-exceptions.md)
* Product: **dotnet-framework**
* GitHub Login: @gewarren
* Microsoft Alias: **dotnetcontent**

### PR DESCRIPTION
**Pull request by jkotas**
Fixes #18616

## Summary

- Clarify that the mapping is not complete
- Add link to the current complete mapping in the runtime sources
- Delete HRESULT mappings that either do not exist or that are corner cases

Fixes #18616
Fixes #18615

### CODE REVIEW

File: docs/framework/interop/how-to-map-hresults-and-exceptions.md [hunk 1]
```diff
@@ -49,36 +49,29 @@ CMyClass::MethodThatThrows
 }  
 ```  
   
- The following table provides the complete mapping from each HRESULT to its comparable exception class in the .NET Framework.  
+ The following table provides the common mappings from HRESULT to its comparable exception class in the .NET.  HRESULT values without explicit mappings are mapped to **COMException**.  The complete up-to-date mapping can be found in [dotnet/runtime repo](https://github.com/dotnet/runtime/blob/master/src/coreclr/vm/rexcep.h).
```
Comment (jkotas) [inline docs/framework/interop/how-to-map-hresults-and-exceptions.md:52]:
```suggestion
 The following table provides the common mappings from HRESULT to its comparable exception class in the .NET. HRESULT values without explicit mappings are mapped to **COMException**.  The complete up-to-date mapping can be found in [dotnet/runtime repo](https://github.com/dotnet/runtime/blob/master/src/coreclr/vm/rexcep.h). 
```

File: docs/framework/interop/how-to-map-hresults-and-exceptions.md [hunk 2]
```diff
@@ -49,36 +49,29 @@ CMyClass::MethodThatThrows
 }  
 ```  
   
- The following table provides the complete mapping from each HRESULT to its comparable exception class in the .NET Framework.  
+ The following table provides the common mappings from HRESULT to its comparable exception class in the .NET. HRESULT values without explicit mappings are mapped to **COMException**.  The complete up-to-date mapping can be found in [dotnet/runtime repo](https://github.com/dotnet/runtime/blob/master/src/coreclr/vm/rexcep.h).
```
Comment (jkotas) [inline docs/framework/interop/how-to-map-hresults-and-exceptions.md:52]:
```suggestion
 The following table provides the common mappings from HRESULT to its comparable exception class in the .NET. HRESULT values without explicit mappings are mapped to **COMException**. The complete up-to-date mapping can be found in [dotnet/runtime repo](https://github.com/dotnet/runtime/blob/master/src/coreclr/vm/rexcep.h). 
```

File: docs/framework/interop/how-to-map-hresults-and-exceptions.md [hunk 3]
```diff
@@ -49,36 +49,29 @@ CMyClass::MethodThatThrows
 }  
 ```  
   
- The following table provides the complete mapping from each HRESULT to its comparable exception class in the .NET Framework.  
+ The following table provides the common mappings from HRESULT to its comparable exception class in the .NET. HRESULT values without explicit mappings are mapped to **COMException**. The complete up-to-date mapping can be found in [dotnet/runtime repo](https://github.com/dotnet/runtime/blob/master/src/coreclr/vm/rexcep.h).
```
Comment (jkotas) [inline docs/framework/interop/how-to-map-hresults-and-exceptions.md:52]: I am not able to figure out what the markdown lint is complaining about.

Comment (adegeo) [reply docs/framework/interop/how-to-map-hresults-and-exceptions.md:52]: It's mad that you have a space at the end of the line, you should have 0. Most articles that were automatically ported added 2 spaces to all lines, which generally get cleaned up when the article gets a full refresh. But 0 is what you should use.

**Review by adegeo — COMMENTED** [review]
I notice that you changed some wording from **.NET Framework** to **.NET**, yet this article was written for framework. Should there also be an article similar to this but for .NET? Perhaps in this area: https://docs.microsoft.com/en-us/dotnet/standard/native-interop/exceptions-interoperability

File: docs/framework/interop/how-to-map-hresults-and-exceptions.md (hunk 3 above)
Comment (jkotas) [reply docs/framework/interop/how-to-map-hresults-and-exceptions.md]: > It's mad that you have a space at the end of the line, you should have 0. 

I have removed the trailing whitespaces from the whole file. Use https://github.com/dotnet/docs/pull/22487/files?w=1 to see the real delta without the whipespace only changes.

**Comment by jkotas** [comment]
This article is equally applicable for .NET Framework and .NET Core. Should it be moved somewhere?

File: docs/framework/interop/how-to-map-hresults-and-exceptions.md [hunk 4]
```diff
@@ -15,127 +15,112 @@ ms.assetid: 610b364b-2761-429d-9c4a-afbc3e66f1b9
...
+    Marshal.ThrowExceptionForHR(COR_E_ARGUMENT);
+}
+```
+
+ The following table provides the common mappings from HRESULT to its comparable exception class in .NET. HRESULT values without explicit mappings are mapped to `COMException`. The complete up-to-date mapping can be found in the [dotnet/runtime repository](https://github.com/dotnet/runtime/blob/master/src/coreclr/vm/rexcep.h).
+
+|HRESULT|.NET exception|
+|-------------|--------------------|
+|`COR_E_APPLICATION`|`ApplicationException`|
+|`COR_E_ARGUMENT or E_INVALIDARG`|`ArgumentException`|
```
Comment (tdykstra) [inline docs/framework/interop/how-to-map-hresults-and-exceptions.md:57]: The "or" shouldn't be included in the code format.  That would prevent it from being localized.
```suggestion
|`COR_E_ARGUMENT` or `E_INVALIDARG`|`ArgumentException`|
```

**Comment by tdykstra** [comment]
>This article is equally applicable for .NET Framework and .NET Core. Should it be moved somewhere?

I believe it would fit better in the [.NET Fundamentals/Native interop TOC](https://github.com/dotnet/docs/blob/868ca7cd5802603d19f02e6e7994511eb8ad873f/docs/fundamentals/toc.yml#L2118).

**Comment by jkotas** [comment]
I would prefer it to be done in separate change. There are multiple articles under framework/interop that are applicable to .NET Core. There may be also a good reason to keep them where they are. While these articles are applicable to .NET Core, they talk about legacy features that were brought over to .NET Core just for compatibility.

**Comment by tdykstra** [comment]
Sounds good. Is this ready to merge, then?

**Comment by jkotas** [comment]
I think so.

**Review by tdykstra — APPROVED** [review]
LGTM

**Pull request merged**